libs/community/langchain_community/tools/azure_ai_services/face_analysis.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, List, Optional
+
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.tools import BaseTool
+from langchain_core.utils import get_from_dict_or_env
+from pydantic import model_validator
+
+logger = logging.getLogger(__name__)
+
+
+class AzureAIFaceAnalysisTool(BaseTool):
+    """
+    A wrapper for the Azure AI Face API.
+
+    This tool facilitates interaction with the Azure AI Face Analysis API,
+    enabling developers to analyze images for facial features and
+    recognition attributes. The analysis can be performed on local image files,
+    with configurable attributes and models for both detection and recognition.
+
+    Attributes:
+        azure_ai_face_key (Optional[str]):
+            The API key for Azure AI Face Analysis.
+        azure_ai_face_endpoint (Optional[str]):
+            The endpoint URL for Azure AI Face Analysis.
+        detection_attributes (Optional[List[str]]):
+            Attributes to be used during face detection.
+        recognition_attributes (Optional[List[str]]):
+            Attributes to be used during face recognition.
+        detection_model (Any):
+            The detection model to use (default: 'DETECTION_03').
+        recognition_model (Any):
+            The recognition model to use (default: 'RECOGNITION_04').
+        formatted_attributes (Any):
+            Formatted attributes to be used during analysis.
+        face_id_time_to_live (int):
+            Time to live for face ID (default: 120 seconds).
+        return_recognition_model (bool):
+            Whether to return recognition model details (default: True).
+        return_face_landmarks (bool):
+            Whether to return face landmarks (default: True).
+        return_face_id (bool):
+            Whether to return face IDs (default: True).
+    """
+
+    azure_ai_face_key: Optional[str] = None  #: :meta private:
+    azure_ai_face_endpoint: Optional[str] = None  #: :meta private:
+    face_client: Any = None  #: :meta private:
+    detection_attributes: Optional[List[str]] = ["HEAD_POSE"]
+    recognition_attributes: Optional[List[str]] = ["QUALITY_FOR_RECOGNITION"]
+    detection_model: Any = "DETECTION_03"
+    recognition_model: Any = "RECOGNITION_04"
+    formatted_attributes: Any = None  #: :meta private:
+    face_id_time_to_live: int = 120
+    return_recognition_model: bool = True
+    return_face_landmarks: bool = True
+    return_face_id: bool = True
+
```

Comment:
We should probably accept an image file path here too, for anyone wanting to invoke the tool directly